firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/CRCElements.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ * Chunks.cpp
+ *
+ *  Created on: Nov 3, 2020
+ *      Author: Renji Panicker
+ *
+ * Unit tests to confirm behavior of chunk splitter
+ *
+ */
+
+#include "Pufferfish/Protocols/CRCElements.h"
+
+#include "Pufferfish/HAL/CRCChecker.h"
+#include "Pufferfish/Test/Util.h"
+#include "Pufferfish/Util/Array.h"
+#include "Pufferfish/Util/Endian.h"
+#include "catch2/catch.hpp"
+
+namespace PF = Pufferfish;
+
+SCENARIO("Protocols::CRCElement behaves correctly", "[CRCElement]") {
+  GIVEN("A CRC element") {
+    constexpr size_t buffer_size = 254UL;
+    using TestCRCElementProps = PF::Protocols::CRCElementProps<buffer_size>;
+    using TestCRCElement = PF::Protocols::CRCElement<TestCRCElementProps::PayloadBuffer>;
+    PF::HAL::SoftCRC32 crc32c{PF::HAL::crc32c_params};
+
+    WHEN("when good data is given as input") {
+      auto crc = PF::Util::make_array<uint8_t>(0x98, 0xdb, 0xe3, 0x55);
+      auto body = std::string("\x98\xdb\xe3\x55\x01\x05\x01\x02\x03\x04\x05", 11);
+      auto expected_payload = std::string("\x01\x05\x01\x02\x03\x04\x05", 7);
+      uint32_t expected_crc = 0;
+      PF::Util::read_ntoh(crc.data(), expected_crc);
```

Comment:
Just for clarity's sake, can we add a `REQUIRE` on what we expect the value of `expected_crc` to be after this function call?

Reply:
https://github.com/pez-globo/pufferfish-software/blob/feature/firmware-test-protocols/firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/CRCElements.cpp#L206 i have added seperate then's for each test with description , is this ok?

Reply:
Looks good!